bundles/org.openhab.binding.logreader/src/main/java/org/openhab/binding/logreader/internal/filereader/FileTailer.java

```diff
@@ -74,9 +76,12 @@ public void start(String filePath, long refreshRate, ScheduledExecutorService sc
     @Override
     public void stop() {
         logger.debug("Shutdown");
-
         if (tailer != null) {
             tailer.stop();
         }
+        if (executor != null) {
+            executor.shutdown();
```

Comment:
Are you sure this is the correct fix? If a user changes configuration dispose will be called an then the stop method here. But this class is still be used. So when start is called again in initialize it will fail as the executor is shutdown. So I think it needs a bit more thought on what is the best approach here. Also there is a `newSingleThreadPool` you can use instead of the fixed with 1.

Reply:
> Are you sure this is the correct fix? If a user changes configuration dispose will be called an then the stop method here. But this class is still be used. So when start is called again in initialize it will fail as the executor is shutdown.

New executor is generated during start.

> Also there is a newSingleThreadPool you can use instead of the fixed with 1.

I actually already used that, but it caused an issue so I rolled back. It gave an exception but can't remember anymore what.

Reply:
@Hilbrand, of course It's also possible to provide dedicated executor from thing handler if you think that is a better solution?